optaplanner-core/src/main/java/org/optaplanner/core/config/heuristic/selector/entity/pillar/SubPillarConfigPolicy.java

```diff
@@ -0,0 +1,106 @@
+        this.subPillarEnabled = true;
+        this.minimumSubPillarSize = minimumSubPillarSize;
+        this.maximumSubPillarSize = maximumSubPillarSize;
+        validateSizes();
+        this.entityComparator = null;
+    }
+
+    private SubPillarConfigPolicy(int minimumSubPillarSize, int maximumSubPillarSize, Comparator<?> entityComparator) {
+        this.subPillarEnabled = true;
+        this.minimumSubPillarSize = minimumSubPillarSize;
+        this.maximumSubPillarSize = maximumSubPillarSize;
+        validateSizes();
+        if (entityComparator == null) {
+            throw new IllegalStateException("entityComparator must not be null.");
+        }
+        this.entityComparator = entityComparator;
+    }
+
+    private SubPillarConfigPolicy() {
+        this.subPillarEnabled = false;
+        this.minimumSubPillarSize = -1;
+        this.maximumSubPillarSize = -1;
+        this.entityComparator = null;
+    }
+
+    public static SubPillarConfigPolicy withoutSubpillars() {
+        return new SubPillarConfigPolicy();
+    }
+
+    public static SubPillarConfigPolicy withSubpillars(int minSize, int maxSize) {
+        return new SubPillarConfigPolicy(minSize, maxSize);
+    }
+
+    public static SubPillarConfigPolicy withSubpillarsUnlimited() {
+        return withSubpillars(1, Integer.MAX_VALUE);
+    }
+
+    public static SubPillarConfigPolicy sequential(int minSize, int maxSize, Comparator<?> entityComparator) {
+        return new SubPillarConfigPolicy(minSize, maxSize, entityComparator);
+    }
+
+    public static SubPillarConfigPolicy sequentialUnlimited(Comparator<?> entityComparator) {
+        return sequential(1, Integer.MAX_VALUE, entityComparator);
+    }
+
+    private void validateSizes() {
+        if (minimumSubPillarSize < 1) {
+            throw new IllegalStateException("The selector (" + this
+                    + ")'s minimumPillarSize (" + minimumSubPillarSize
+                    + ") must be at least 1.");
+        }
+        if (minimumSubPillarSize > maximumSubPillarSize) {
+            throw new IllegalStateException("The minimumPillarSize (" + minimumSubPillarSize
+                    + ") must be at least maximumSubChainSize (" + maximumSubPillarSize + ").");
+        }
+    }
+
+    public boolean isSubPillarEnabled() {
+        return subPillarEnabled;
+    }
+
+    /**
+     * @return Less than 1 when {@link #isSubPillarEnabled()} false.
+     */
+    public int getMinimumSubPillarSize() {
+        return minimumSubPillarSize;
+    }
+
+    /**
+     * @return Less than 1 when {@link #isSubPillarEnabled()} false.
+     */
+    public int getMaximumSubPillarSize() {
+        return maximumSubPillarSize;
+    }
+
+    /**
+     * @return Only present if the subpillars are to be treated as sequential. Never present if
+     * {@link #isSubPillarEnabled()} is false.
+     */
+    public Optional<Comparator<?>> getEntityComparator() {
```

Comment:
Let's not use Optional. They don't make the code easier to read. Optionals are for Stream purposes.
Just rreturn the Compartor and have the javadoc state ist's null if subpillarenable is false

Reply:
I'm going to resist the urge to bikeshed now. But we will inevitably get back to this at some later point, since I'm not going to let you take `Optional` away from me so easily. :-)